src/pip/_internal/commands/show.py

```diff
@@ -166,3 +179,60 @@ def print_results(distributions, list_files=False, verbose=False):
             if "files" not in dist:
                 logger.info("Cannot locate installed-files.txt")
     return results_printed
+
+
+def print_json(distributions, list_files=False, verbose=False):
+    """
+    Print in JSON format the information from installed distributions found.
+    """
+    results_printed = False
+    for i, dist in enumerate(distributions):
+        results_printed = True
+        if i > 0:
+            logger.info("---")
```

Comment:
Why would you want that ?
I'd expect the `--json` option to output valid json: a list of dict containing the metadata.